### PR TITLE
chore: add context and address some TODOs in noxfile

### PR DESCRIPTION
**Pull request by tswast**
Part of fix-it week.
🦕

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/core/generic.py [hunk 1]
```diff
@@ -167,7 +167,7 @@ def astype(self, dtype):
             dtype (str or pandas.ExtensionDtype):
                 A dtype supported by BigQuery DataFrame include ``'boolean'``, ``'Float64'``, ``'Int64'``,
                 ``'int64[pyarrow]'``, ``'string'``, ``'string[pyarrow]'``, ``'timestamp[us, tz=UTC][pyarrow]'``,
-                ``'timestamp\[us\]\[pyarrow\]'``, ``'date32\[day\]\[pyarrow\]'``, ``'time64\[us\]\[pyarrow\]'``.
+                ``'timestamp[us][pyarrow]'``, ``'date32[day][pyarrow]'``, ``'time64[us][pyarrow]'``.
```
Comment (junyazhang) [inline third_party/bigframes_vendored/pandas/core/generic.py:170]: There is a bug in the doc generation, which it renders incorrectly without the escape symbol: https://screenshot.googleplex.com/45wRMR3yQUuuGdB

With the escape symbol: https://screenshot.googleplex.com/7uGCXAoC6coMRoC

**Review by junyazhang — APPROVED** [review]
Thank you Tim for addressing the TODOs!

File: third_party/bigframes_vendored/pandas/core/generic.py (hunk 1 above)
Comment (tswast) [reply third_party/bigframes_vendored/pandas/core/generic.py:170]: Bummer. That's really annoying. `\[` is invalid in a Python string, but equivalent to `\\[`. I'll update the text.

Comment (tswast) [reply third_party/bigframes_vendored/pandas/core/generic.py]: Fixed in https://github.com/googleapis/python-bigquery-dataframes/pull/572/commits/1e438d25bc71c446aa5b508223fba123957c13f1

**Comment by tswast** [comment]
Marking as `do not merge` until after the release today.

**Pull request merged**